homeassistant/components/kraken/sensor.py

```diff
@@ -0,0 +1,182 @@
+"""The kraken integration."""
+from datetime import timedelta
+import logging
+
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import DATA_UPDATED, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=5)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add kraken entities from a config_entry."""
+    kraken_data = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors = []
+
+    for asset_pair in kraken_data.tradable_asset_pairs:
+        sensors.append(
+            KrakenSensor(
+                kraken_data, config_entry.data["name"], asset_pair[0], asset_pair[1]
+            )
+        )
+    async_add_entities(sensors, True)
+
+
+class KrakenSensor(Entity):
+    """Define a Kraken sensor."""
+
+    def __init__(self, kraken_data, name, pair_name, pair):
+        """Initialize."""
+        self._kraken_data = kraken_data
+        self._pair_name = pair_name
+        self._unit_of_measurement = pair[1]
+        self._name = "_".join([name, *pair])
+        self._state = None
+        self._available = True
+        self._async_remove_dispatcher = None
+        self._attrs = {
+            "ask": None,
+            "ask_volume": None,
+            "bid": None,
+            "bid_volume": None,
+            "volume_today": None,
+            "volume_last_24h": None,
+            "volume_weighted_average_today": None,
+            "volume_weighted_average_last_24h": None,
+            "number_of_trades_today": None,
+            "number_of_trades_last_24h": None,
+            "low_today": None,
+            "low_last_24h": None,
+            "high_today": None,
+            "high_last_24h": None,
+            "opening_price_today": None,
```

Comment:
Please break all of these attributes out to individual sensor entities instead. If some of them are less relevant we can disable them by default.